File: app/simple/Gallery/src/main/kotlin/com/onyx/gallery/action/shape/ShapeChangeAction.java
package com.onyx.gallery.action.shape;

import androidx.annotation.NonNull;

import com.onyx.android.sdk.rx.RxCallback;
import com.onyx.android.sdk.scribble.shape.ShapeFactory;
import com.onyx.gallery.common.BaseEditAction;
import com.onyx.gallery.request.shape.UpdateCurrentShapeTypeRequest;

/**
 * <pre>
 *     author : liao lin tao
 *     time   : 2018/9/6 18:22
 *     desc   :
 * </pre>
 */
public class ShapeChangeAction extends BaseEditAction {
    private int shapeType = ShapeFactory.SHAPE_BRUSH_SCRIBBLE;

    public ShapeChangeAction setShapeType(int shapeType) {
        this.shapeType = shapeType;
        return this;
    }

    @Override
    public void execute(RxCallback rxCallback) {
        UpdateCurrentShapeTypeRequest request = new UpdateCurrentShapeTypeRequest(shapeType);
        getNoteManager().enqueue(request, new RxCallback<UpdateCurrentShapeTypeRequest>() {
            @Override
            public void onNext(@NonNull UpdateCurrentShapeTypeRequest updateCurrentShapeTypeRequest) {
                RxCallback.onNext(rxCallback, request);
            }

            @Override
            public void onError(Throwable e) {
                super.onError(e);
                RxCallback.onError(rxCallback, e);
            }
        });
    }
}

File: app/simple/Gallery/src/main/kotlin/com/onyx/gallery/action/shape/ShapeChangeAction.kt
package com.onyx.gallery.action.shape

import com.onyx.android.sdk.rx.RxCallback
import com.onyx.android.sdk.rx.RxRequest
import com.onyx.android.sdk.scribble.shape.ShapeFactory
import com.onyx.gallery.bundle.EditBundle
import com.onyx.gallery.common.BaseEditAction
import com.onyx.gallery.request.shape.UpdateCurrentShapeTypeRequest

/**
 * Created by Leung 2020/7/17 18:05
 **/
class ShapeChangeAction(editBundle: EditBundle, var shapeType: Int = ShapeFactory.SHAPE_BRUSH_SCRIBBLE) : BaseEditAction<RxRequest>(editBundle) {

    override fun execute(rxCallback: RxCallback<RxRequest>?) {
        editBundle.enqueue(UpdateCurrentShapeTypeRequest(editBundle, shapeType), object : RxCallback<RxRequest>() {
            override fun onNext(request: RxRequest) {
                RxCallback.onNext(rxCallback, request)
            }

            override fun onError(e: Throwable) {
                super.onError(e)
                RxCallback.onError(rxCallback, e)
            }
        })
    }
}